file.cc

```diff
@@ -269,102 +163,41 @@ bool Class::unmap( uchar * address )
 }
 
 
-void Class::seekCur( qint64 offset ) 
+void Class::seekEnd()
 {
-  if ( writeBuffer )
-    flushWriteBuffer();
-
-  if( !f.seek( f.pos() + offset ) )
-    throw exSeekError();
-}
-
-void Class::seekEnd( qint64 offset ) 
-{
-  if ( writeBuffer )
-    flushWriteBuffer();
-
-  if( !f.seek( f.size() + offset ) )
+  if ( !f.seek( f.size()) )
     throw exSeekError();
```

Comment:
what does seekEnd do ,the original code `f.size()+offset` looks suspicious.

Reply:
Yes, the original `seekEnd` is wrong. seek() beyond end is undefined.

However, this behavior has never used in anywhere. The default value of `offset` is `0` and this value is never changed. So there is no bug, and I removed the parameter.

Reply:
![image](https://user-images.githubusercontent.com/105986/232041874-9f730d71-18ae-479c-899f-7c8fab66244f.png)

maybe intended , If gd want to reserve some space for extra data at the end